compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/ExternalDrag.desktop.kt

```diff
@@ -0,0 +1,474 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.State
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberUpdatedState
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.AwtWindowDragTargetListener.WindowDragState
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.geometry.Rect
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.layout.boundsInWindow
+import androidx.compose.ui.layout.onGloballyPositioned
+import androidx.compose.ui.window.LocalWindow
+import androidx.compose.ui.window.density
+import java.awt.Point
+import java.awt.Window
+import java.awt.dnd.DnDConstants
+import java.awt.dnd.DropTarget
+import java.awt.dnd.DropTargetDragEvent
+import java.awt.dnd.DropTargetDropEvent
+import java.awt.dnd.DropTargetEvent
+import java.awt.dnd.DropTargetListener
+
+/**
+ * Represent data that is being dragged (or dropped) to a component from outside an application.
+ */
+@ExperimentalComposeUiApi
+interface DragData {
+    /**
+     * List of all MIME types for this [DragData].
+     * The list is ordered from most richly descriptive to least descriptive.
+     *
+     * Some platform-specific mimeTypes can occur in the list. It may be changed in future versions.
+     */
+    val mimeTypes: List<String>
```

Comment:
It seems if we can't read an arbitrary byte array, we don't need to know mime types?

If it is only needed to know if we dosn't support it, we can just use `else` branch when we check the type.

Reply:
Hm, agree. I thought about providing a way to show UI like "Please, DnD Image" when user dragged smth else, but you are right that it can be done by checking the needed type